source/postprocess/rotation_statistics.cc

```diff
@@ -0,0 +1,260 @@
+/*
+  Copyright (C) 2011 - 2018 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/rotation_statistics.h>
+#include <aspect/material_model/simple.h>
+#include <aspect/global.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace
+    {
+      void add_scientific_column(const std::string &name,
+                                 const double value,
+                                 TableHandler &statistics)
+      {
+        statistics.add_value(name,value);
+        statistics.set_precision (name, 8);
+        statistics.set_scientific (name, true);
+      }
+    }
+
+    template <int dim>
+    std::pair<std::string,std::string>
+    RotationStatistics<dim>::execute (TableHandler &statistics)
```

Comment:
I wonder whether  the function would be easier to read if you had specializations for the 2d and 3d cases separately. Opinions?